packages/react/src/generators/setup-ssr/setup-ssr.ts

```diff
@@ -0,0 +1,175 @@
+
+export async function setupSsrGenerator(tree: Tree, options: Schema) {
+  const projectConfig = readProjectConfiguration(tree, options.project);
+  const projectRoot = projectConfig.root;
+  const appImportCandidates = [
+    options.appComponentImportPath,
+    'app',
+    'App',
+    'app/App',
+    'App/App',
+  ];
+  const appComponentImport = appImportCandidates.find(
+    (app) =>
+      tree.exists(joinPathFragments(projectConfig.sourceRoot, `${app}.tsx`)) ||
+      tree.exists(joinPathFragments(projectConfig.sourceRoot, `${app}.jsx`)) ||
+      tree.exists(joinPathFragments(projectConfig.sourceRoot, `${app}.js`))
+  );
+
+  if (!appComponentImport) {
+    throw new Error(
+      `Cannot find an import path for <App/> component. Try passing setting --appComponentImportPath option.`
+    );
+  }
+
+  if (!projectConfig.targets.build || !projectConfig.targets.serve) {
+    throw new Error(
+      `Project ${options.project} does not have build and serve targets`
+    );
+  }
+
+  if (projectConfig.targets.server) {
+    throw new Error(`Project ${options.project} already has a server target.`);
+  }
+
+  const originalOutputPath = projectConfig.targets.build?.options?.outputPath;
+
+  if (!originalOutputPath) {
+    throw new Error(
+      `Project ${options.project} does not contain a outputPath for the build target.`
+    );
+  }
+
+  projectConfig.targets.build.options.outputPath = joinPathFragments(
+    originalOutputPath,
+    'browser'
+  );
+  projectConfig.targets = {
+    ...projectConfig.targets,
+    server: {
+      executor: '@nrwl/webpack:webpack',
+      outputs: ['{options.outputPath}'],
+      defaultConfiguration: 'production',
+      options: {
+        target: 'node',
+        main: `${projectRoot}/server.ts`,
+        outputPath: joinPathFragments(originalOutputPath, 'server'),
+        tsConfig: `${projectRoot}/tsconfig.server.json`,
+        compiler: 'babel',
+        externalDependencies: 'all',
+        outputHashing: 'none',
+        webpackConfig: '@nrwl/react/plugins/webpack',
+      },
+      configurations: {
+        development: {
+          optimization: false,
+          sourceMap: true,
+        },
+        production: {
+          fileReplacements: [
+            {
+              replace: `${projectRoot}/src/environments/environment.ts`,
+              with: `${projectRoot}/src/environments/environment.prod.ts`,
+            },
+          ],
+          sourceMap: false,
+        },
+      },
+    },
+    'serve-browser': projectConfig.targets.serve,
+    'serve-server': {
+      executor: '@nrwl/js:node',
+      defaultConfiguration: 'development',
+      options: {
+        buildTarget: `${options.project}:server:development`,
+        buildTargetOptions: {
+          watch: true,
+        },
+      },
+      configurations: {
+        development: {},
+        production: {
+          buildTarget: `${options.project}:server:production`,
+        },
+      },
+    },
+    serve: {
```

Comment:
It may not be a big deal, but in Angular we maintain `serve` as pointing to the browser build, and have a `serve-ssr` for running browser + server. We don't have a single `serve-server` to just serve the server.

So we align these? I followed the nguniversal approach of having `serve` and `serve-ssr` along with a `server` target with the last being the actual build for the server.

I don't mind, it can be cleanup, but not sure if we want the same DX across both or not

Reply:
I see. I think it's not useful to have `serve` only do CSR. Especially if the server provides endpoints needed by the client (like backend-for-frontend).